kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/MetricsIT.java

```diff
@@ -324,6 +353,98 @@ void shouldIncrementCountOnMessageTransit(UnaryOperator<ConfigurationBuilder> bu
+                                .filterByTag(NODE_ID_LABEL, "bootstrap")
+                                .singleElement()
+                                .value()
+                                .isGreaterThan(0.0)),
+                argumentSet("count node connection from proxy to server ",
+                        (UnaryOperator<ConfigurationBuilder>) builder -> builder,
+                        (Consumer<List<SimpleMetric>>) metricList -> assertThat(metricList)
+                                .filterByName("kroxylicious_proxy_to_server_connections_total")
+                                .filterByTag(NODE_ID_LABEL, "0")
+                                .isEmpty(),
+                        (Consumer<List<SimpleMetric>>) metricList -> assertThat(metricList)
+                                .filterByName("kroxylicious_proxy_to_server_connections_total")
+                                .filterByTag(NODE_ID_LABEL, "0")
+                                .singleElement()
+                                .value()
+                                .isGreaterThan(0.0)),
+                argumentSet("count error from proxy to server)",
+                        (UnaryOperator<ConfigurationBuilder>) builder -> builder,
+                        (Consumer<List<SimpleMetric>>) metricList -> assertThat(metricList)
+                                .filterByName("kroxylicious_proxy_to_server_errors_total")
+                                .filterByTag(NODE_ID_LABEL)
+                                .isEmpty(),
+                        (Consumer<List<SimpleMetric>>) metricList -> assertThat(metricList).anySatisfy(simpleMetric -> {
+                            assertThat(simpleMetric.name()).isEqualTo("kroxylicious_proxy_to_server_errors_total");
+                            assertThat(simpleMetric.value()).isGreaterThan(0);
+                        })));
+    }
+
+    @ParameterizedTest
+    @MethodSource("connectionMetricsScenarios")
+    void shouldIncrementCountOnConnectionMetrics(UnaryOperator<ConfigurationBuilder> builder,
+                                                 Consumer<List<SimpleMetric>> beforeAssertion,
+                                                 Consumer<List<SimpleMetric>> afterAssertion,
+                                                 KafkaCluster cluster)
+            throws Exception {
+        var config = configWithMetrics(cluster);
+
+        // Given
+        try (var tester = kroxyliciousTester(builder.apply(config));
+                var managementClient = tester.getManagementClient();) {
+
+            var metricList = managementClient.scrapeMetrics();
+
+            beforeAssertion.accept(metricList);
+
+            // When
+            var producer = tester.producer();
+            var future = producer.send(new ProducerRecord<>("my-topic", "key", "value"));
+            cluster.close();
+            assertThat(future).succeedsWithin(Duration.ofSeconds(10));
```

Comment:
The same issue with this test as well, the test fails on the case where we check for `kroxylicious_proxy_to_server_errors_total` metrics. This tests passes locally but fails here